### PR TITLE
nomad: nomad_1_6 -> nomad_1_7, nomad_1_7: 1.7.2 -> 1.7.3. nomad_1_6: 1.6.5 -> 1.6.6, nomad_1_5: 1.5.12 -> 1.5.13, nomad_1_4: remove

### PR DESCRIPTION
**Pull request by cottand**
## Description of changes

- Updated patch versions of Nomad 1.5, 1.6, 1.7
- Changed `nomad` to point to the latest Nomad version

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by amaxine** [comment]
Thanks! This would also be a good time to drop `nomad_1_4`.

**Comment by Cottand** [comment]
Done! I have never removed a package before so not sure what the convention is. For now I followed [this](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md).

File: pkgs/applications/networking/cluster/nomad/default.nix [hunk 1]
```diff
@@ -52,22 +51,15 @@ rec {
...
 
-  nomad_1_4 = generic {
-    buildGoModule = buildGo120Module;
-    version = "1.4.12";
-    sha256 = "sha256-dO98FOaO5MB5pWzeF705s/aBDTaF0OyWnVxWGB91suI=";
-    vendorHash = "sha256-D5TcTZa64Jr47u4mrTXK4lUIC5gfBQNVgL6QKh1CaQM=";
-    license = lib.licenses.mpl20;
-    passthru.tests.nomad = nixosTests.nomad;
-  };
+  nomad_1_4 = throw "nomad_1_4 was removed - you can use nomad_1_5 or an older nixpkgs revision";
```
Comment (techknowlogick) [inline pkgs/applications/networking/cluster/nomad/default.nix:56]: Thanks for PR, could you include this in changelog?

**Comment by amaxine** [comment]
Thanks - we've usually not added aliases about nomad version removals, or release notes in fact, but I think that was probably a mistake - thanks for doing that!

Small nit for the message: `nomad_1_4 is no longer supported upstream. You can switch to using a newer version of the nomad package, or revert to older nixpkgs if you cannot upgrade` or something along those lines.

I don't think this warrants a release note, but the update to Nomad 1.7 as default probably does - even if there's a decent chance 1.8 will be the default by May :slightly_smiling_face:

**Comment by Cottand** [comment]
Hey @amaxine and @techknowlogick - thanks for your reviews. I think they are both breaking changes so I added both to the changelog

**Comment by techknowlogick** [comment]
Thanks @Cottand <3

Would you be able to rebase on master? Otherwise the merge commit in your branch will be included when this PR is merged.

If you need any help with rebasing or resolving conflicts please feel free to ping and I can help :)

**Comment by Cottand** [comment]
Rebased!

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**